src/core/Akka/Util/ByteIterator.cs

```diff
@@ -0,0 +1,471 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ActorBase.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+//     Copyright (C) 2013-2015 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Akka.IO;
+
+namespace Akka.Util
+{
+    internal abstract class ByteIterator
+    {
+        internal class ByteArrayIterator : ByteIterator
+        {
+            private byte[] _array;
+            private int _until;
+            private int _from;
+
+            public ByteArrayIterator(byte[] array, int @from, int until)
+            {
+                _array = array;
+                _from = @from;
+                _until = until;
+            }
+
+            public override int Len
```

Comment:
While this is actually direct compatibility from Akka, my vote is to get rid of abbreviations - we should keep close to C# naming conventions in C# code. cc/ @akkadotnet/developers